lib/ansible/modules/database/influxdb/influxdb_user.py

```diff
@@ -179,12 +235,21 @@ def main():
                     client.revoke_admin_privileges(user_name)
                     changed = True
             except influx.exceptions.InfluxDBClientError as e:
-                module.fail_json(msg=str(e))
+                module.fail_json(msg=to_native(e))
 
-            module.exit_json(changed=changed)
         else:
             user_password = user_password or ''
             create_user(module, client, user_name, user_password, admin)
+            changed = True
+
+        if grants:
```

Comment:
The default to an empty list may result in a unexpected behaviour, if users don't use this param (default empty list) and have set grants manually, the grants would get deleted with this implementation. 

I would change it in a way to only handle it if a list (be it empty) is set and ignore it if None (param unset):
~~~python
        if grants is not None:
            if set_user_grants(module, client, user_name, grants):
                changed = True
~~~

Reply:
It's done, thanks